pytmc/xml_collector.py

```diff
@@ -874,72 +736,122 @@ def valid(self):
                                   for key in self._required_fields)
         return has_required_keys and has_required_fields
 
-    def render_record(self):
+    @property
+    def records(self):
+        """Generated :class:`.EPICSRecord` objects"""
+        return NotImplemented
+
+    def render_records(self):
         """
         Returns
         -------
         string
-            Jinja rendered entry for this BaseRecordPackage
+            Jinja rendered entry for the RecordPackage
         """
-        simple_dict = self.cfg_as_dict()
         if not self.valid:
-            logger.error('Unable to render record: %s', simple_dict)
+            logger.error('Unable to render record: %s', self)
             return
+        return '\n'.join([record.render_template()
```

Comment:
Maybe?

```suggestion
        return '\n\n'.join([record.render_template().strip()
```

That is, ignore any whitespace in the template at the beginning or end, and ensure two line breaks between each record.